common/oauth.go

```diff
@@ -164,8 +165,8 @@ func (w *observableTokenSource) HasChanged(tok *oauth2.Token) bool {
 		return true
 	}
 
-	return w.lastKnown.AccessToken == tok.AccessToken &&
+	return !(w.lastKnown.AccessToken == tok.AccessToken &&
```

Comment:
fixing a bug, token was not updating because of this.

Reply:
Nice catch!

As an optimization, the following is equivalent and more readable (doesn't require a large expression to be put into parenthesis with a `!` in front of it.)

```
          return w.lastKnown.AccessToken != tok.AccessToken ||
		w.lastKnown.RefreshToken != tok.RefreshToken ||
		w.lastKnown.TokenType != tok.TokenType ||
		!w.lastKnown.Expiry.Equal(tok.Expiry)
```